### PR TITLE
Discover pilot parameters (Commands, extensions and version) at runtime.

### PR DESCRIPTION
**Pull request by CinziaLu**
Discover pilot parameters (Commands, extensions and version) at runtime.

Removed the -X and -E command line options.
The json file is not created if it already exists.

For issue #2724

### CODE REVIEW

File: Pilot/pilotTools.py [hunk 1]
```diff
@@ -74,9 +74,10 @@ def retrieveUrlTimeout( url, fileName, log, timeout = 0 ):
       expectedBytes = 0
     data = remoteFD.read()
     if fileName:
-      localFD = open( fileName + '-local', "wb" )
-      localFD.write( data )
-      localFD.close()
+      if not os.path.exists( fileName + '-local' ):
+        localFD = open( fileName + '-local', "wb" )
```
Comment (fstagni) [inline Pilot/pilotTools.py]: Please use the `with` construct:

```
with open( fileName + '-local', "wb" ) as localFD:
  localFD.write( data )
```

File: Pilot/tests/Test_Pilot.py [hunk 2]
```diff
@@ -30,14 +30,26 @@ def test_GetPilotVersion( self ):
...
+    json.dump( {'TestSetup':{'Commands':{'grid1':['x', 'y', 'z'], 'grid2':['d', 'f']}, 'Version':['v1r1', 'v2r2']}}, fp )
     fp.close()
     self.pp.setup = 'TestSetup'
     self.pp.pilotCFGFileLocation = 'file://%s' % os.getcwd()
     gpv = GetPilotVersion( self.pp )
     self.assertTrue( gpv.execute() is None )
     self.assertEqual( gpv.pp.releaseVersion, 'v1r1' )
 
+  def test_RetrievePilotParameters( self ):
+    fp = open( 'pilot.json', 'w' )
```
Comment (fstagni) [inline Pilot/tests/Test_Pilot.py]: You can use `with` also here

File: Pilot/pilotTools.py [hunk 3]
```diff
@@ -553,4 +546,41 @@ def __initOptions( self ):
...
+      import json
+      self.log.info( "Finding the pilot commands list" )
+      result = retrieveUrlTimeout( self.pilotCFGFileLocation + '/' + self.pilotCFGFile,
+                                   self.pilotCFGFile,
+                                   self.log,
+                                   timeout = 120 )
+      if not result:
+        self.log.info( "Could not download the json file, using the default commands list." )
+      else:
+        fp = open( self.pilotCFGFile + '-local', 'r' )
```
Comment (fstagni) [inline Pilot/pilotTools.py]: ... and `with` also here

File: Pilot/pilotCommands.py [hunk 4]
```diff
@@ -359,6 +367,117 @@ def _getSecurityCFG( self ):
       self.cfg.append( "-o /DIRAC/Security/CertFile=%s/hostcert.pem" % self.pp.certsLocation )
       self.cfg.append( "-o /DIRAC/Security/KeyFile=%s/hostkey.pem" % self.pp.certsLocation )
 
+class CheckCECapabilities( CommandBase ):
```
Comment (fstagni) [inline Pilot/pilotCommands.py]: I don't understand: why this command (and the next) were not already present?

Comment (CinziaLu) [inline Pilot/pilotCommands.py]: I do not know, in DIRAC v6r15 is there. Maybe @wkrzemien can comment.

**Pull request merged**